tpls/mdspan/include/experimental/__p2630_bits/submdspan_extents.hpp

```diff
@@ -17,6 +17,7 @@
 #pragma once
 
 #include <tuple>
+#include <complex>
```

Comment:
Are you sure you want/need to include `<complex>` here ? It's surprising.

Reply:
See https://github.com/kokkos/mdspan/pull/358/files#r1742703222.

Reply:
But isn't this pair-like thingy used for slicing ? If so, I would be surprise that slicing is allowed with a complex... I can't image how bad a code would be if complex numbers were used for that.

Reply:
It is by the standard.

Reply:
turns out that the ISO C++ standard defines the exposition only concept pair-like which is defined through tuple-like which is defined here: https://eel.is/c++draft/tuple.like#concept:tuple-like

So `complex<double>` is a valid slice specifier for `submdspan` because its pair-like and double is convertible to `index_type`

Reply:
It is a draft right ? If so I would refrain supporting such a ill-formed case.

Reply:
no this is final. This was discussed in the committee and it is basically not avoidable. We want "convertible to index_type" to be valid, that means floating points will be valid. And the committee did not feel like adding an explicit "convertible to index_type but not floating point" to the specification of all these things in mdspan.

Reply:
OK, then I wouldn't go against the committee.

But in this specific case, you're still including `complex` in a very central file, so any downstream project using `Kokkos` would have `complex` included ?!

Reply:
That is already the case, because you get it indirectly through Kokkos_Complex

Reply:
I still think that `Kokkos` should not pull `<xomplex>` for me, but do what you think is best. I won't block for this of course :wink: